packages/component-library-react/TESTING.md

```diff
@@ -0,0 +1,207 @@
+
+```jsx
+it("can have a additional class name", () => {
+  const { container } = render(<MyComponent className="large" />);
+
+  const component = container.querySelector(":only-child");
+
+  expect(component).toHaveClass("large");
+  expect(component).toHaveClass("my-component");
+});
+```
+
+## Test the accessibility tree
+
+### Landmarks
+
+```jsx
+it("renders an complementary role element", () => {
+  render(<Aside />);
+
+  const aside = screen.getByRole("complementary");
+
+  expect(aside).toBeInTheDocument();
+});
+```
+
+### Label for landmarks
+
+Some components have an API to configure the label:
+
+```jsx
+it("renders an complementary role element with a name", () => {
+  render(<BreadcrumbNav label="Breadcrumbs" />);
+
+  const nav = screen.getByRole("navigation", { name: "Breadcrumbs" });
+
+  expect(nav).toBeInTheDocument();
+});
+```
+
+Other components need to rely on `aria-labelledby` or `aria-label`.
+
+```jsx
+it('renders an complementary role element with a name', () => {
+  render(
+    <Aside aria-labelledby="heading">
+      <h2 id="heading">See also</h1>
+    </Aside>
+  );
+
+  const aside = screen.getByRole('complementary', { name: 'See also' });
+
+  expect(aside).toBeInTheDocument();
+});
+```
+
+### States
+
+Voor [WCAG 4.1.2](https://nldesignsystem.nl/wcag/4.1.2) is het belangrijk dat de state van componenten beschikbaar is in de accessibility tree. [Testing Library heeft APIs](https://testing-library.com/docs/queries/byrole) om de informatie uit de accessibility tree op te vragen, in plaats van via de DOM.
+
+Voorbeelden van state zijn:
+
+- Een `checkbox` die `checked` is.
+- Een `textbox` die `disabled` is.
+- Een `textarea` die `required` is.
+- Een `button` die `expanded` is.
+
+```jsx
+describe("checked variant", () => {
+  it("is not checked by default", () => {
+    const { container } = render(<Checkbox />);
+
+    const checkbox = screen.getByRole("checkbox");
+
+    expect(checkbox).not.toBeChecked();
+  });
+
+  it("can have a checked state", () => {
+    const handleChange = () => {};
+    render(<Checkbox checked onChange={handleChange} />);
+
+    const checkbox = screen.getByRole("checkbox");
+
+    expect(checkbox).toBeChecked();
+  });
+
+  it("can have a defaultChecked state (in React)", () => {
+    render(<Checkbox defaultChecked />);
+
+    const checkbox = screen.getByRole("checkbox");
+
+    expect(checkbox).toBeChecked();
+  });
+});
+```
+
+Helaas ondersteunt Testing Library nog niet elke state in de accessibility tree. Maak alvast wel de test, maar sla de test over `skip`. Gebruik de DOM om de test op een alternative manier te doen.
+
+```jsx
+// `aria-disabled` is somehow not recognized as disabled state
+it.skip("has a disabled listbox", () => {
+  render(<Listbox disabled />);
+
+  const listbox = screen.getByRole("listbox");
+
+  expect(listbox).toBeDisabled();
+});
```

Comment:
@Robbert `aria-disabled` niet kunnen testen via de `toBeDisabled()` handler is een bewuste keuze van Testing Library. Ik denk ook dat het niet aan te raden is om standaard een skipped test te schrijven hiervoor, aangezien dat de verwachting wekt dat de test ooit wel klaar is voor gebruik. Het lijkt mij handiger om expliciet te testen wat de gevolgen moeten zijn van het meegeven van de disabled prop (oftewel het hebben van `aria-disabled`).

Reply:
@bddjong Goeie tip! Heb je het over deze discussie?

https://github.com/testing-library/jest-dom/issues/144

Reply:
@bddjong Ik heb de documentatie, de voorbeelden en de tests getweakt op basis van je feedback